### ISSUE TITLE
[java] Parse error on rare extra dimensions on method return type on annotation methods

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3642]
**Affects PMD Version:** 6.40.0

Make sure, to test with the latest PMD version.

**Description:** title. Found in checkstyle sources in #3640 

**Code Sample demonstrating the issue:**

```java
@interface Foo {
    String v1()[]; // parse error
    // equivalent to String[] v1();
}
```

**Comment by adangel** [comment]
Interesting: the eclipse parser also shows an parse error... but it seems to be valid java code...

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
